vsintegration/src/FSharp.Editor/CodeFixes/CodeFixHelpers.fs

```diff
@@ -106,3 +108,57 @@ module internal CodeFixExtensions =
                 let! sourceText = ctx.GetSourceTextAsync()
                 return RoslynHelpers.TextSpanToFSharpRange(ctx.Document.FilePath, ctx.Span, sourceText)
             }
+
+// This cannot be an extension on the code fix context
+// because the underlying GetFixAllProvider method doesn't take the context in.
+#nowarn "3511" // state machine not statically compilable
+
+[<AutoOpen>]
+module IFSharpCodeFixProviderExtensions =
+    type IFSharpCodeFixProvider with
+
+        member provider.RegisterFsharpFixAll() =
+            FixAllProvider.Create(fun fixAllCtx doc allDiagnostics ->
+                cancellableTask {
+                    let sw = Stopwatch.StartNew()
+
+                    let! token = CancellableTask.getCurrentCancellationToken ()
+                    let! sourceText = doc.GetTextAsync token
+
+                    // this is not used anywhere, it's just needed to create the context
+                    let action =
+                        Action<CodeActions.CodeAction, ImmutableArray<Diagnostic>>(fun _ _ -> ())
```

Comment:
If it's not used, it should be a constant, to not allocate action every single time.

Reply:
Done